localstack/services/stepfunctions/asl/component/state/state_execution/state_task/state_task.py

```diff
@@ -59,6 +59,41 @@ def from_state_props(self, state_props: StateProps) -> None:
         self.parameters = state_props.get(Parameters)
         self.resource = state_props.get(Resource)
 
+    def _get_supported_parameters(self) -> Optional[set[str]]:  # noqa
+        return None
+
+    def _get_parameters_normalising_bindings(self) -> dict[str, str]:  # noqa
+        return dict()
+
+    def _eval_parameters(self, env: Environment) -> dict:
+        # Eval raw parameters.
+        parameters = dict()
+        if self.parameters:
+            self.parameters.eval(env=env)
+            parameters = env.stack.pop()
+
+        # Handle supported parameters.
+        supported_parameters = self._get_supported_parameters()
+        if supported_parameters:
+            unsupported_parameters: list[str] = [
+                parameter
+                for parameter in parameters.keys()
+                if parameter not in supported_parameters
+            ]
+            for unsupported_parameter in unsupported_parameters:
+                parameters.pop(unsupported_parameter, None)
+
+        # Normalise bindings.
+        parameter_normalisers = self._get_parameters_normalising_bindings()
+        for parameter_key in list(parameters.keys()):
```

Comment:
what's the reason here for wrapping it with `list()`?

Reply:
I don't want the list to mutate during iteration, so I save a reference to its current state just before starting. It can change with api actions or executions.